src/methods/clipping/clipping_processor.jl

```diff
@@ -0,0 +1,295 @@
+# # This file contains the shared helper functions for the other polygon clipping
+# # functionalities.
+
+"""
+    _poly_in_poly(::GI.PolygonTrait, poly_a, ::GI.PolygonTrait, poly_b):: Bool 
+
+    This function is only called when we know the two polygons do not intersect. This
+    returns true if poly_a is completely contained in poly_b and false otherwise.
+"""
+# TODO: tempory _poly_in_poly, only works if entirely contained
+# Checks if polya in poly b
+function _poly_in_poly(poly_a, poly_b)
+    p_a = to_edges(poly_a)[1][1]
+    return point_in_polygon(p_a, poly_b)
+end
+
+"""
+    _my_get_pts(::GI.PolygonTrait, poly_a)::Vector{Tuple{Float64}}
+
+    The GeometryOps to_points was not working so I wrote this one. It just
+    used the 'to_edges' function from GeometryOps to derive to_points. 
+"""
+# TODO: temporary alternative for to_points, repeats first and last point
+function _my_get_pts(poly_a)
+    edges = to_edges(poly_a)
+    len =  length(edges) + 1
+    poly_points = Vector{Tuple{Float64, Float64}}(undef, len)
+    for ii in eachindex(edges)
+        poly_points[ii] = edges[ii][1]
+    end
+    poly_points[end] = edges[1][1]
+    return poly_points
+end
+
+"""
+    _lin_ring_to_poly(lin_ring)::GI.Polygon
+
+    This function turns a linear ring into a GeometryInterface polygon.
+"""
+function _lin_ring_to_poly(lin_ring)
+    edges = to_edges(lin_ring)
+    points = Vector{Tuple{Float64, Float64}}(undef, length(edges)+1)
+    for ii in eachindex(edges)
+        points[ii] = edges[ii][1]
+    end
+    points[end] = edges[1][1]
+    return GI.Polygon([points])
+end
+
+"""
+    _build_a_list(
+        ::GI.PolygonTrait, 
+        poly_a, ::GI.PolygonTrait, poly_b,
+         edges_a, edges_b, intr_list, a_idx_list, 
+         b_idx_list, alpha_a_list, alpha_b_list, 
+         a_list, b_list
+    )::intr_list, a_idx_list, b_idx_list, alpha_a_list, alpha_b_list, counter
+
+    This function take in two polygons, the lists of their edges, and some lists containing
+    information about the nature of their intersection points. From that, it creates an 
+    array to represent polygon_a, and this array contains intersection points. However, after
+    calling this function, a_list is not fully formed because the neighboring indicies of the
+    intersection points in b_list still need to be updated. We will have fully formed a_list
+    after calling _build_ab_list. Also at this point we still have not update the entry and
+    exit flags for a_list.
+"""
+function _build_a_list(::GI.PolygonTrait, poly_a, ::GI.PolygonTrait, poly_b, edges_a, edges_b,
```

Comment:
Since we aren't dispatching for these helper functions (all of the build lists), you don't need the polygon traits as arguments. We only want that if we are planning on having different function calls for different kinds of shapes.